Meme-ify-Me/app/src/main/java/nyc/c4q/rosmaryfc/meme_ify_me/MainActivity.java

```diff
@@ -95,6 +91,40 @@ private void takePhoto (View v){
         startActivityForResult(intent, TAKE_PICTURE);
     }
 
+    //method for gathering intent information from takePhoto and pickPhoto methods
+    // and setting the imageview with correct bitmap
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data){
+        if(resultCode == RESULT_OK ){
+                if (requestCode == PICK_PICTURE){
+                    selectedImagePath = String.valueOf(data.getData());
+                    imageview.setImageBitmap(decodePhoto(selectedImagePath));
+                } else if (requestCode == TAKE_PICTURE){
+                    selectedImagePath = imageUri.toString();
+                    imageview.setImageBitmap(decodePhoto(selectedImagePath));
+                } else {
+                    super.onActivityResult(requestCode, resultCode, data);
+                }
+        }
+    }
```

Comment:
I like that you guys are sharing code, makes this less repetitive and more efficient. In my branch, I also changed your take photo and pick photo methods from private to public so that I could use it with android: onClick